### PR TITLE
Fix potential one-definition-rule breaking structs

### PR DESCRIPTION
**Pull request by binji**
Context is the most common potential problem here, but I noticed that
wasm-link.h had some others too (DataSegment and Export).

These probably aren't an issue yet, since they all have trivial
constructors. As soon as we add non-trivial types to these they break
though.

### CODE REVIEW

File: src/ast-writer.cc [hunk 1]
```diff
@@ -69,6 +71,8 @@ struct Context {
   int func_type_index;
 };
 
+}  // namespace
```
Comment (sbc100) [inline src/ast-writer.cc]: Is there a reason not to put these outside the wabt namespace?

Comment (binji) [inline src/ast-writer.cc]: Could do, shouldn't really make a difference either way. Is there a benefit to not having it nested?

**Comment by binji** [comment]
Looks like the Travis error is a flaky git fetch.

File: src/ast-writer.cc (hunk 1 above)
Comment (KarlSchimpf) [inline src/ast-writer.cc]: Using the anonymous namespace is standard practice in C++ for static variables, since all variables in the anonymous namespace is automatically static.

However, I forgot to notice you should remove the "static" because the name is in the anonymous namespace (again, standard C++ practice).

Comment (binji) [inline src/ast-writer.cc]: Agreed, though that's a bigger change. Mind if I do it in another PR?

Comment (KarlSchimpf) [inline src/ast-writer.cc]: No problem commit as is.
